src/components/ApplyFiltersModal/index.tsx

```diff
@@ -42,27 +48,60 @@ const ApplyFiltersModal: React.FC<ApplyFiltersModalProps> = ({
   const { project } = data || {};
   const { parsleyFilters } = project || {};
 
+  const onConfirm = () => {
+    // Apply selected filters.
+    const newFilters = filters.concat(state.selectedFilters);
+    setFilters(newFilters);
+
+    // Send relevant tracking events.
+    leaveBreadcrumb(
+      "applied-default-filters",
+      { filters: state.selectedFilters },
+      "user"
+    );
+    sendEvent({
+      name: "Applied Default Filters",
+      filters: state.selectedFilters,
+    });
+    setOpen(false);
+    dispatch({ type: "RESET" });
+  };
+
+  const onCancel = () => {
+    setOpen(false);
+    dispatch({ type: "RESET" });
+  };
+
   return (
     <ConfirmationModal
       buttonText="Apply filters"
       css={css`
         z-index: ${zIndex.modal};
       `}
-      onCancel={() => setOpen(false)}
-      onConfirm={() => setOpen(false)}
+      data-cy="apply-filters-modal"
+      onCancel={onCancel}
+      onConfirm={onConfirm}
       open={open}
       setOpen={setOpen}
+      submitDisabled={state.selectedFilters.length === 0}
       title="Default Filters"
     >
       <Scrollable>
         {parsleyFilters?.map((f) => (
           <DefaultFilter
             key={f.expression}
             activeFilters={filters}
+            addFilter={(filterToAdd: Filter) =>
```

Comment:
```suggestion
            addFilter={(filterToAdd) =>
```

nit. This should already be typed from the prop